### PR TITLE
Rowlog tool for diagnosing vreplication issues

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
Tool developed while researching a issue faced during a migration. More details in the doc which is part of this PR

Signed-off-by: Rohit Nayak <rohit@planetscale.com>

### CODE REVIEW

**Comment by rohit-nayak-ps** [comment]
Just noticed that there are unrelated files in the PR from the time I was testing this. Converting to draft until I fix it.

File: examples/local/scripts/vttablet-up.sh [hunk 1]
```diff
@@ -47,8 +47,7 @@ vttablet \
  -health_check_interval 5s \
  -enable_semi_sync \
  -enable_replication_reporter \
- -heartbeat_enable \
- -heartbeat_interval 250ms \
+ -heartbeat_enable=false \
```
Comment (shlomi-noach) [inline examples/local/scripts/vttablet-up.sh:50]: `master` branch already has these removed. seems to me like an update from `master` is useful.

File: go/vt/vttablet/tabletserver/tabletserver.go [hunk 2]
```diff
@@ -215,7 +215,7 @@ func (tsv *TabletServer) InitDBConfig(target querypb.Target, dbcfgs *dbconfigs.D
 	tsv.txThrottler.InitDBConfig(target)
 	tsv.vstreamer.InitDBConfig(target.Keyspace)
 	tsv.hs.InitDBConfig(target)
-	tsv.lagThrottler.InitDBConfig(target.Keyspace, target.Shard)
+	//tsv.lagThrottler.InitDBConfig(target.Keyspace, target.Shard)
```
Comment (shlomi-noach) [inline go/vt/vttablet/tabletserver/tabletserver.go:218]: Please uncomment. Throttler now disabled by default. (merge `master`)

File: tools/rowlog/rowlog.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+###rowlog
```
Comment (shlomi-noach) [inline tools/rowlog/rowlog.md:1]:
```suggestion
# rowlog
```

File: tools/rowlog/rowlog.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+`rowlog` is a diagnostic tool for researching issues reported by users in vreplication workflows. These are usually
+reported as data mismatches or loss and are not reproducible because of the huge amounts of data involved and 
+data privacy reasons. Also, the database configurations vary greatly: number/type/size of columns, collations, 
+keys/indexes etc.
+
+In order to rule out that the problem is not related to the core vreplication algorithm this tool has been built.
+
+`rowlog` is a standalone command-line tool. For a given set of primary keys for a table (only non-composite 
+integer primary keys are supported) it outputs the binlogs associated with these ids, both on the source and
```
Comment (shlomi-noach) [inline tools/rowlog/rowlog.md:11]: > (only non-composite integer primary keys are supported) 

Is there a plan to support arbitrary keys?

**Review by shlomi-noach — COMMENTED** [review]
Very exciting and useful tool

File: go/vt/vttablet/tabletserver/tabletserver.go (hunk 2 above)
Comment (rohit-nayak-ps) [reply go/vt/vttablet/tabletserver/tabletserver.go:218]: Actually I converted this to a draft PR and is superseded by https://github.com/vitessio/vitess/pull/6892. These throttler related changes were not going to be part of the release.

**Comment by rohit-nayak-ps** [comment]
Closing since it is superseded by https://github.com/vitessio/vitess/pull/6892.

**Pull request closed without merging**